### PR TITLE
Hide community momitoring module

### PR DESCRIPTION
**Pull request by hacheG**

### CODE REVIEW

File: src/app/layout/header/Menu.jsx [hunk 1]
```diff
@@ -68,11 +68,13 @@ class Menu extends React.Component {
               </li>
             </Link>
           )}
+          {showAlerts && (
```
Comment (erikasv) [inline src/app/layout/header/Menu.jsx:71]: oh no, the whole point is to create a new variable, not to reuse the alerts one

File: src/pages/home/Content.jsx [hunk 2]
```diff
@@ -65,6 +65,8 @@ const Content = ({ activeModule, setActiveModule }) => {
       auth: false,
     });
   }
+
+  if (showAlert) {
```
Comment (erikasv) [inline src/pages/home/Content.jsx:69]: same comment as in Menu component.

Also fix indentation

File: src/App.jsx [hunk 3]
```diff
@@ -133,7 +133,7 @@ class App extends React.Component {
             <Route path="/GEB/Compensaciones" component={this.loadCompensator} />
             <Route path="/Portafolios" render={this.loadPortfolio} />
             <Route path="/Alertas" render={this.loadHome} />
-            <Route path="/Monitoreo" render={this.loadCBMDashboard} />
+            <Route path="/Monitoreo" render={this.loadHome} />
```
Comment (erikasv) [inline src/App.jsx:136]: Add a conditional here or in the loadCBMDashboard function, so in develop it still show the CBM module

**Review by erikasv — CHANGES_REQUESTED** [review]

File: src/App.jsx [hunk 4]
```diff
@@ -133,7 +133,7 @@ class App extends React.Component {
             <Route path="/GEB/Compensaciones" component={this.loadCompensator} />
             <Route path="/Portafolios" render={this.loadPortfolio} />
             <Route path="/Alertas" render={this.loadHome} />
-            <Route path="/Monitoreo" render={this.loadCBMDashboard} />
+            <Route path="/Monitoreo" render={process.env.REACT_APP_ENVIRONMENT === "develop" ? this.loadCBMDashboard : this.loadHome} />
```
Comment (erikasv) [inline src/App.jsx:136]: no, this shouldn't be available based on the env var, but on the feature flag.

File: src/app/layout/header/Menu.jsx [hunk 5]
```diff
@@ -11,12 +11,16 @@ class Menu extends React.Component {
...
+      showCBMDashboard: false
     };
   }
 
   componentDidMount() {
     isFlagEnabled('alertsModule')
       .then((value) => this.setState({ showAlerts: value }));
+    
+      isFlagEnabled('CBMModule')
+      .then((value) => this.setState({ showCBMDashboard: value }));
```
Comment (erikasv) [inline src/app/layout/header/Menu.jsx:23]: fix indentation

File: src/app/layout/header/Menu.jsx [hunk 6]
```diff
@@ -68,11 +72,13 @@ class Menu extends React.Component {
               </li>
             </Link>
           )}
+          {showCBMDashboard && (
```
Comment (erikasv) [inline src/app/layout/header/Menu.jsx:75]: fix inner indentation

File: src/pages/home/Content.jsx [hunk 7]
```diff
@@ -9,12 +9,18 @@ import isFlagEnabled from 'utils/isFlagEnabled';
...
+  const [showCBMDashboard, setshowCBMDashboard] = useState(false);
 
   useEffect(() => {
     isFlagEnabled('alertsModule')
       .then((value) => setShowAlert(value));
   }, []);
 
+  useEffect(() => {
+    isFlagEnabled('CBMModule')
+      .then((value) => setshowCBMDashboard(value));
```
Comment (erikasv) [inline src/pages/home/Content.jsx:21]: this should go in the previous useEffect (it doesn't have different conditions)

**Review by erikasv — CHANGES_REQUESTED** [review]

File: src/App.jsx (hunk 4 above)
Comment (hacheG) [reply src/App.jsx:136]: hola, entiendo lo de no dejar aqui la variable de ambiente, pero me podrias aclarar esta correccion, osea por la correccion anterior sobre este mismo punto se que debo dejar que se pueda acceder a la ruta desde develop pero en esta correccion me podrias por fa explicar un poco mas

Comment (erikasv) [reply src/App.jsx:136]: tienes que comparar con el valor de la feature flag, no con el de la variable de ambiente

Comment (hacheG) [reply src/App.jsx:136]: :D graciaaas!!!!

File: src/App.jsx [hunk 8]
```diff
@@ -15,17 +15,23 @@ import 'main.css';
...
     this.state = {
       user: null,
       headerNames: {},
+      showCBMDashboard: false
     };
   }
 
   buildQuery = (queryString) => new URLSearchParams(queryString);
 
+  setShowCBMDashboard = isFlagEnabled('CBMModule')
```
Comment (erikasv) [inline src/App.jsx:32]: 1. setShowCBMDashboard is not being used
2. setState doesn't return anything, setShowCBMDashboard will be a promise that resolves to undefined.
3. The state update must go inside componentDidMount function (if it doesn't exist create it).

the rest of the variables directly declared in the class are functions, this mean they aren't executed until they are needed, in this case setShowCBMDashboard could not have finished before the value of showCBMDashboard is used

File: src/App.jsx [hunk 9]
```diff
@@ -94,7 +100,7 @@ class App extends React.Component {
   );
 
   loadCBMDashboard = () => (
-    this.loadComponent({
+      this.loadComponent({
```
Comment (erikasv) [inline src/App.jsx:103]: undo this indentation

File: src/App.jsx [hunk 10]
```diff
@@ -133,7 +139,7 @@ class App extends React.Component {
             <Route path="/GEB/Compensaciones" component={this.loadCompensator} />
             <Route path="/Portafolios" render={this.loadPortfolio} />
             <Route path="/Alertas" render={this.loadHome} />
-            <Route path="/Monitoreo" render={this.loadCBMDashboard} />
+            <Route path="/Monitoreo" render={showCBMDashboard ? this.loadCBMDashboard: this.loadHome} />
```
Comment (erikasv) [inline src/App.jsx:142]: add space before `:`

File: src/app/layout/header/Menu.jsx [hunk 11]
```diff
@@ -61,18 +65,20 @@ class Menu extends React.Component {
               Portafolios
             </li>
           </Link>
-          {showAlerts && (
+          {!showAlerts && (
```
Comment (erikasv) [inline src/app/layout/header/Menu.jsx:68]: why did you add a negation (`!`)?

File: src/app/layout/header/Menu.jsx [hunk 12]
```diff
@@ -61,18 +65,20 @@ class Menu extends React.Component {
...
                 Alertas
               </li>
             </Link>
           )}
-          <Link to="/Monitoreo" onClick={this.changeMenuState}>
-            <li>
-              Monitoreo comunitario
-            </li>
-          </Link>
+          {!showCBMDashboard && (
```
Comment (erikasv) [inline src/app/layout/header/Menu.jsx:75]: the condition is inverted. Or why the negation(`!`)?

File: src/pages/home/Content.jsx [hunk 13]
```diff
@@ -54,7 +58,7 @@ const Content = ({ activeModule, setActiveModule }) => {
     },
   ];
 
-  if (showAlert) {
+  if (!showAlert) {
```
Comment (erikasv) [inline src/pages/home/Content.jsx:61]: why the negation?

File: src/pages/home/Content.jsx [hunk 14]
```diff
@@ -65,15 +69,18 @@ const Content = ({ activeModule, setActiveModule }) => {
...
-    focusCallback: () => setActiveModule('cbmdashboard'),
-    buttonStyles: `finder mon ${(activeModule === 'cbmdashboard') ? 'activeicon' : ''}`,
-    idBtn: 'monbtn',
-    firstLineContent: 'Monitoreo',
-    secondLineContent: 'comunitario',
-    localLink: '/Monitoreo',
-    auth: false,
-  });
+
+  if (!showCBMDashboard) {
```
Comment (erikasv) [inline src/pages/home/Content.jsx:73]: why the negation?

**Review by erikasv — CHANGES_REQUESTED** [review]

File: src/pages/home/Content.jsx [hunk 15]
```diff
@@ -53,7 +57,7 @@ const Content = ({ activeModule, setActiveModule }) => {
       auth: false,
     },
   ];
-
+console.log("SHOW_ALERT - SHOW_CBM",showAlert, showCBMDashboard);
```
Comment (erikasv) [inline src/pages/home/Content.jsx:60]: remove console.log

**Review by erikasv — APPROVED** [review]

**Pull request merged**